libmamba/include/mamba/core/util.hpp

```diff
@@ -131,18 +218,25 @@ namespace mamba
 
         static bool is_locked(const LockFile& lockfile)
         {
+            return lockfile.is_locked() &&
 #ifdef _WIN32
-            return is_locked(lockfile.lockfile_path());
+                   is_locked(lockfile.lockfile_path());
```

Comment:
Why do we need this double check, not only `lockfile.is_locked()` ?

Reply:
Because this function needs to check if the path is blocked "in general", not only blocked by this process. Checking `lockfile.is_locked()` returns true only if the current process is locking that path, if not, we need to check if some other process locks it.